storage/scylla_api.c

```diff
@@ -315,34 +314,57 @@ static status_t get_blob(const CassRow* row, cass_byte_t* target, const char* co
   return ret;
 }
 
-static status_t create_permanent_keyspace(CassSession* session) {
+static char* make_query(const char* head_desc, const char* position, const char* left_desc) {
+  if (head_desc == NULL || position == NULL || left_desc == NULL) {
+    return NULL;
+  }
+  size_t head_len = strlen(head_desc);
+  size_t pos_len = strlen(position);
+  size_t left_len = strlen(left_desc);
+  size_t result_len = head_len + pos_len + left_len + 1;
+  char* result = malloc(result_len * sizeof(char));
+  if (result == NULL) {
+    return NULL;
+  }
+  memcpy(result, head_desc, head_len);
+  memcpy(result + head_len, position, pos_len);
+  memcpy(result + head_len + pos_len, left_desc, left_len);
+  result[result_len - 1] = 0;
+
+  return result;
+}
+
+static status_t create_permanent_keyspace(CassSession* session, const char* keyspace_name) {
+  if (keyspace_name == NULL) {
```

Comment:
I wonder if the feasibility when `keyspace_name` == null. I don't see the necessity. Can you clarify?

Reply:
I think `keyspace_name` could only be checked in function `make_query` where it actually will be used and should ensure it is not NULL.
I'd like to revise `make_query` from `static char* make_query( ... )` to `static status_t make_query(char** query, ... )` to get error code(TA_NULL or OOM)